### PR TITLE
Data type additions

### PR DESCRIPTION
**Pull request by mmbell**
This PR adds two new data types, renames one type, and removes some debugging messages. 

The new data types are NOAA HRD thinned Doppler radials that are sent off the P3 aircraft in real-time with the suffix .list, and HDOBS in situ data format produced by both NOAA and USAF aircraft with the suffix .hdob.

The 'model' data type previously had a suffix of .txt, but this caused some confusion and user errors since the suffix was not descriptive enough and is frequently used with files that are not in the prescribed format. The suffix has been changed to .model which is not commonly in use and is more descriptive. Samurai will now ignore .txt files rather than try to read them in.

Some debugging messages that printed on loop when loading a background file were removed. 

A commit to match up the CMakeLists.txt file with the current master version that has been approved in multiple recent PRs was also added in order to avoid conflicts and enable a seamless merge of this branch.

### CODE REVIEW

**Comment by johnmauff** [comment]
@mmbell  This PR is missing a number of changes made by myself and Jian that were made weeks ago which prevents me from easily verifying that the code actually runs correctly on both CPU and GPU.  I am guessing that you did not update your hrd_radius branch with commits from the master.  Could you do this?  Otherwise, all of the changes that Jian and myself have made will be overwritten.

**Review by johnmauff — CHANGES_REQUESTED** [review]
Missing a very large number of critical commits to the master branch.

**Review by tingyucha — APPROVED** [review]
Looks good to me!

**Comment by mmbell** [comment]
@johnmauff You were correct. I thought that the PR would only affect the files I modified, but I was behind on master. I added two commits to update my local master and merge it in with hrd_radials. The history now looks accurate to me and has all of your's and Jian's commits. Please review that I did it correctly and then we can merge it. Thanks!

**Review by johnmauff — APPROVED** [review]
Michael, thanks for updating your branch.  I was able to verify that your modifications do not change the results on the GPU.  I did not verify this on the CPU but based on the code changes, I am confident that this will be true as well.

File: src/BkgdObsLoaders.cpp [hunk 1]
```diff
@@ -558,7 +558,7 @@ bool BkgdObsSplineLoader::loadBkgdObs(std::vector<real> &bgIn)
     real rhoBar = refstate->getReferenceVariable(ReferenceVariable::rhoaref, heightm);
     real qBar = refstate->getReferenceVariable(ReferenceVariable::qvbhypref, heightm);
     real tBar = refstate->getReferenceVariable(ReferenceVariable::tempref, heightm);
-		std::cout << "tBar = " << tBar << std::endl;
+		//std::cout << "tBar = " << tBar << std::endl;
```
Comment (sjsprecious) [inline src/BkgdObsLoaders.cpp:561]: I would prefer removing this line than commenting it out.

File: src/BkgdObsLoaders.cpp [hunk 2]
```diff
@@ -603,7 +603,7 @@ bool BkgdObsSplineLoader::loadBkgdObs(std::vector<real> &bgIn)
 		}
 
     // Push values for an entire column, then solve for the spline
-		std::cout << "logheights size = " << logheights.size() << std::endl;
+		//std::cout << "logheights size = " << logheights.size() << std::endl;
```
Comment (sjsprecious) [inline src/BkgdObsLoaders.cpp:606]: I would prefer removing this line than commenting it out.

File: src/BkgdObsLoaders.cpp [hunk 3]
```diff
@@ -955,7 +955,10 @@ bool BkgdObsKDLoader::loadBkgdObs(std::vector<real> &bgIn)
     std::cout << "*** Debug KD Step: " << debugKdStep << std::endl;
   }
 
+    //int lineno = 0;
   while( bkgdAdapter->next(time, lat, lon, alt, u, v, w, t, qv, rhoa, qr, terrain_hgt) ) {
+      //lineno = lineno +1;
+      //std::cout << "*** Debug Lineno: " << lineno << std::endl;
```
Comment (sjsprecious) [inline src/BkgdObsLoaders.cpp:961]: I would prefer removing these lines than commenting them out.

**Review by sjsprecious — APPROVED** [review]
Looks good to me. Just a couple of minor comments.

I will issue a separate PR to incorporate the new CMake changes into the NCAR build/run scripts later.

File: src/BkgdObsLoaders.cpp [hunk 4]
```diff
@@ -955,7 +955,10 @@ bool BkgdObsKDLoader::loadBkgdObs(std::vector<real> &bgIn)
     std::cout << "*** Debug KD Step: " << debugKdStep << std::endl;
   }
 
+    //int lineno = 0;
```
[the remark this answers is not in the document]
Comment (mmbell) [reply src/BkgdObsLoaders.cpp]: I agree. I have removed those lines from the file.

**Comment by mmbell** [comment]
All changes reviewed and accepted.

**Pull request merged**